hack/run-locally.sh

```diff
@@ -94,11 +101,64 @@ function wait_for_cluster() {
     fi
 }
 
+function from_scratch() {
+    if [[ -n "${INSTALLER_DIR:-}" ]]; then
+        return 0
+    fi
+    return 1
+}
+
+PLUGIN_IMAGE=${PLUGIN_IMAGE:-}
+NETWORK_PLUGIN="OpenShiftSDN"
+IMAGE_ENV_KEY="NODE_IMAGE"
+CLUSTER_DIR=${CLUSTER_DIR:-}
+INSTALLER_DIR=${INSTALLER_DIR:-}
+while getopts "c:i:m:n:" opt; do
+    case $opt in
+        c) CLUSTER_DIR=${OPTARG};;
+        i) INSTALLER_DIR=${OPTARG};;
+        m) PLUGIN_IMAGE=${OPTARG};;
+        n)
+            case ${OPTARG} in
+                ovn)
+                    NETWORK_PLUGIN="OVNKubernetes"
+                    IMAGE_ENV_KEY="OVN_IMAGE"
+                    ;;
+                sdn|"")
+                    ;;
+                *)
+                    echo "Unknown network plugin '${OPTARG}'" >&2
+                    exit 1
+                ;;
+            esac
+            ;;
+        \?)
+            echo "Invalid option: -${OPTARG}" >&2
+            exit 1
+            ;;
+        :)
+            echo "Option -${OPTARG} requires an argument." >&2
+            exit 1
+            ;;
+    esac
+done
+
 if [[ -z "${CLUSTER_DIR:-}" ]]; then
-    echo "error: CLUSTER_DIR must be set"
+    echo "error: CLUSTER_DIR must be set or '-c <cluster-dir>' must be given"
     echo "For more info, see INSTALLER-HACKING.md"
     exit 1
 fi
+mkdir -p ${CLUSTER_DIR} >& /dev/null
+
+if from_scratch ; then
+    echo "Creating install config..."
+    ${INSTALLER_DIR}/bin/openshift-install --dir=${CLUSTER_DIR} create install-config
```

Comment:
That kind of looks like you're expecting the caller to pass the path to the installer source tree... People are not expected to build the installer from source any more unless they are hacking on the installer. You're supposed to download a pre-built binary.

You could take a path to the `openshift-install` binary itself instead. And make it optional and just run it from `$PATH` if it's not specified.

Reply:
done.